classes/order/Order.php

```diff
@@ -2448,6 +2451,17 @@ public function getProductTaxesDetails($limitToOrderDetails = false)
 
                 $product_specific_discounts[$cart_rule->reduction_product] += $order_cart_rule['value_tax_excl'];
                 $order_discount_tax_excl -= $order_cart_rule['value_tax_excl'];
+            } elseif ($cart_rule->gift_product > 0) {
```

Comment:
This would need a behat test to make sure the behavior is not broken again in the future

It should probably be added here in: `tests/Integration/Behaviour/Features/Scenario/Order/order_gift_cart_rule.feature` 👍